test/scripts/parallel-local/main.js

```diff
@@ -0,0 +1,234 @@
+ * @param {string} testSuiteName - The name of the test suite directory.
+ * @param {string} logFileName - The name of the log file to read.
+ * @returns {void} Does not return a value.
+ */
+const readLogFiles = (testSuiteName, logFileName) => {
+    const testDirectoryPath = path.join('./database/storage/', testSuiteName);
+    try {
+        const files = fs.readdirSync(testDirectoryPath);
+        files.forEach((file) => {
+            if (file === logFileName) {
+                const filePath = path.join(testDirectoryPath, file);
+                const data = fs.readFileSync(filePath, 'utf8');
+                console.log(data);
+            }
+        });
+    } catch (err) {
+        console.error(`Error reading directory for ${testSuiteName}: ${err}`);
+    }
+};
+
+/**
+ * Aggregates results by counting amount of occurrences of passed, failing, and pending tests.
+ * @param {string} testSuiteName - The name of the test suite.
+ * @param {string} logFileName - The log file from which to read and count results.
+ * @param {string} searchKeyword - The keyword to search for in the log entries (e.g., "Passing").
+ * @returns {number} The total count of occurrences of the search keyword.
+ */
+const aggregateResults = (testSuiteName, logFileName, searchKeyword) => {
+    const testDirectoryPath = path.join('./database/storage/', testSuiteName);
+    try {
+        const files = fs.readdirSync(testDirectoryPath);
+        const file = files.find((file) => file === logFileName);
+        if (file) {
+            const filePath = path.join(testDirectoryPath, file);
+            const data = fs.readFileSync(filePath, 'utf8');
+            return data.split('\n')
+                .map((line) => line.trim()) // Remove any leading or trailing whitespace
+                .filter((line) => line.endsWith(searchKeyword)) // Ensure the line ends with the keyword
+                .reduce((acc, line) => {
+                    const parts = line.split(' '); // Split the line into parts
+                    const count = parseInt(parts[0], 10); // The number is the first part of the line
+                    return acc + (isNaN(count) ? 0 : count); // Add it to the accumulator ensuring it's a number
+                }, 0);
+        }
+    } catch (err) {
+        console.error(`Error reading directory for ${testSuiteName}: ${err}`);
+    }
+    return 0;
+};
+
+/**
+ * Cleans up directories for each test suite after tests have completed.
+ * This function iterates through each test suite and removes its corresponding directory,
+ * ensuring the testing environment is clean for subsequent runs.
+ *
+ * @param {Array<string>} testSuites - Array of test suite names whose directories are to be cleaned up.
+ * @returns {void} Does not return a value.
+ */
+const cleanupTestDirectories = (testSuites) => {
+    testSuites.forEach((testSuiteName) => {
+        const testDirectoryPath = path.join('./database/storage/', testSuiteName);
```

Comment:
This should be either configured using environment variable (best) or at least a constant

Reply:
Done, chose constant for now